src/index.spec.ts

```diff
@@ -382,6 +394,167 @@ describe('EppoJSClient E2E test', () => {
   });
 });
 
+describe('decoupled initialization', () => {
+  let mockLogger: IAssignmentLogger;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  let init: (config: IClientConfig) => Promise<EppoJSClient>;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  let getInstance: () => EppoJSClient;
+
+  beforeEach(async () => {
+    jest.isolateModules(() => {
+      // Isolate and re-require so that the static instance is reset to its default state
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const reloadedModule = require('./index');
+      init = reloadedModule.init;
+      getInstance = reloadedModule.getInstance;
+    });
+  });
+
+  describe('isolated from the singleton', () => {
+    beforeEach(() => {
+      mockLogger = td.object<IAssignmentLogger>();
+
+      global.fetch = jest.fn(() => {
+        const ufc = { flags: { [obfuscatedFlagKey]: mockObfuscatedUfcFlagConfig } };
+
+        return Promise.resolve({
+          ok: true,
+          status: 200,
+          json: () => Promise.resolve(ufc),
+        });
+      }) as jest.Mock;
+    });
+
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it('should be independent of the singleton', async () => {
+      const apiOptions: IApiOptions = { sdkKey: '<MY SDK KEY>' };
+      const options: IClientOptions = { ...apiOptions, assignmentLogger: mockLogger };
+      const isolatedClient = new EppoJSClient(options);
```

Comment:
do we want people calling constructors or should we have some method that does this? My spidey senses tell me the later, but that may be left over bias from the Java community's war on constructors a decade ago

Reply:
The main reason to prefer builder methods is that they make it easier to maintain backward compatibility. If constructor is private, we can fundamentally change how construction works without breaking API. We can also add builder methods for common setups and give them meaningful names

Reply:
moved to a builder